### ISSUE TITLE
JavaScript Reader attempts to dispatch all messages even after poll is terminated

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3275]
The receiver should do the same isTerminated() check the others are doing.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3374
Reporter: narupley
Created: 2014-07-21T08:30:19.000-0700

**Comment by rbeckman-nextgen** [comment]
Fixed in revision 7235.

Imported Comment. Original Details:
Author: narupley
Created: 2014-07-24T17:28:39.000-0700

**Comment by rbeckman-nextgen** [comment]
Created a channel using a JavaScript Reader as the source connector, added an ArrayList with over 600 messages and logger statements for when the poll begins and just before it returns to the preprocessor. Finally, I added a Thread.sleep method in the preprocessor script.

After deploying the channel in 3.0.3, I stopped and halted the channel which caused errors to occur for each subsequent message that hadn't been dispatched. I exported and imported this channel into 3.1 and repeated the same steps. No errors were logged this time.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-09-03T11:03:23.000-0700
